src/tdx/azure/azure_tdx_adapter.c

```diff
@@ -665,15 +685,15 @@ int json_unmarshal_quote_response(char **quote,
 		goto ERROR;
 	}
 
-	char* tmp_string = (char *)json_string_value(tmp_obj);
-	size_t quote_size;
-	if (json_string_length(tmp_obj) % 4 != 0) {
-		quote_size = json_string_length(tmp_obj) + (4 - (json_string_length(tmp_obj) % 4));
-	} else {
-		quote_size = json_string_length(tmp_obj);
-	}
+	char *tmp_string = (char *)json_string_value(tmp_obj);
+	size_t tmp_length = json_string_length(tmp_obj);
+
+	// Calculate the size for the quote string, including any padding
+	size_t padding_needed = (4 - tmp_length % 4) % 4; // Number of '=' needed for padding
+	size_t quote_size = tmp_length + padding_needed;  // Add the padding to the length
+
+	*quote = (char *)calloc(quote_size + 1, sizeof(char));
```

Comment:
This function calloc(...) may be guarded by checking the variable tmp_length to avoid allocating trivial memory space for invalidated tmp_obj.

Reply:
fixed in https://github.com/intel/trustauthority-client-for-c/pull/2/commits/a2d97b9041133f22f9d8ae30fb0c4b43e2f6327a